images/operator/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# syntax=docker/dockerfile:1.1-experimental
+ARG CILIUM_BUILDER_IMAGE=docker.io/errordeveloper/cilium-builder:019538f3bd3275e8d8369e2c5823efbe0f3a4f06-dev
+ARG CA_CERTIFICATES_IMAGE=docker.io/cilium/ca-certificates:69a9f5d66ff96bf97e8b9dc107e92aa9ddbdc9a8
+
+FROM --platform=linux/amd64 ${CILIUM_BUILDER_IMAGE} as builder
+
+ARG NOSTRIP
+ARG LOCKDEBUG
+
+RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
+  make -C operator \
+    NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG PKG_BUILD=1 \
+  && mkdir -p /out/linux/amd64/usr/bin && mv operator/cilium-operator* /out/linux/amd64/usr/bin
+
+RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
+  env GOARCH=arm64 \
+    make -C operator \
+      NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG PKG_BUILD=1 \
+  && mkdir -p /out/linux/arm64/usr/bin && mv operator/cilium-operator* /out/linux/arm64/usr/bin
+
+
+FROM ${CA_CERTIFICATES_IMAGE}
+ARG TARGETPLATFORM
+LABEL maintainer="maintainer@cilium.io"
+
+COPY --from=builder /out/${TARGETPLATFORM} /
+
+WORKDIR /
+CMD ["/usr/bin/cilium-operator-generic"]
```

Comment:
what's the relationship between this dockerfile and the current per-provider docker image building?

Reply:
To be honest, I'm just trying to keep it simple here, this image gets all 4 binaries, but I'd like to find a way to have separate images without having separate (largely duplicate) Dockerfiles per-se. Part of me is really keen to explore abstracting the Dockerfiles into some simple YAML format and generating Dockerfile on the fly, or using syntax extension mechanism that [BuildKit now offers](https://matt-rickard.com/building-a-new-dockerfile-frontend/).